### PR TITLE
Add an iterator to prevent OOM when cleaning large databases

### PR DESCRIPTION
**Pull request by craigmaloney**
## Description
Adds an iterator to .all() to iterate over the data rather than load it all in memory.

## Related Issue
#603 

## Motivation and Context
Prevents loading database completely into memory.

## How Has This Been Tested?
Tested in a VM with 6GB of memory that was previously (and consistently) giving an OOM error

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] I have run the `make format` command to format my code
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [X] I have added my name and/or github handle to `AUTHORS.rst`
- [X] I have added my change to `CHANGES.rst`
- [X] All new and existing tests passed.

### CODE REVIEW

File: simple_history/management/commands/clean_duplicate_history.py [hunk 1]
```diff
@@ -69,7 +69,7 @@ def _process(self, to_process, date_back=None, dry_run=True):
             # have changes (in the given period) but
             # `m_qs.values(model._meta.pk.name).distinct()`
             # is actually slower than looping all and filtering in the code...
-            for o in model.objects.all():
+            for o in model.objects.all().iterator():
```
Comment (rossmechanic) [inline simple_history/management/commands/clean_duplicate_history.py]: do you need `.all()` or can you just use `.iterator()`? Also, was just looking at the docs. Will this make a separate db query for every for loop? Or am I reading that wrong?

**Comment by craigmaloney** [comment]
My understanding is that the iterator will get 2,000 records at a time and keep a cursor open on the database.

It appears the .all().iterator() can be trimmed to .iterator(). I'll make that change.

File: simple_history/management/commands/clean_duplicate_history.py (hunk 1 above)
Comment (craigmaloney) [reply simple_history/management/commands/clean_duplicate_history.py]: My understanding is that the iterator will get 2,000 records at a time and keep a cursor open on the database.

It appears the .all().iterator() can be trimmed to .iterator(). I'll make that change.

File: simple_history/management/commands/clean_duplicate_history.py [hunk 2]
```diff
@@ -69,7 +69,7 @@ def _process(self, to_process, date_back=None, dry_run=True):
             # have changes (in the given period) but
             # `m_qs.values(model._meta.pk.name).distinct()`
             # is actually slower than looping all and filtering in the code...
-            for o in model.objects.all():
+            for o in model.objects..iterator():
```
Comment (rossmechanic) [inline simple_history/management/commands/clean_duplicate_history.py]: oops! Looks like you left the extra `.` in there. Fix that and I'll approve and merge

**Comment by craigmaloney** [comment]
Bah. Sorry about that. Removed the typo. Thanks!

File: simple_history/management/commands/clean_duplicate_history.py (hunk 2 above)
Comment (craigmaloney) [reply simple_history/management/commands/clean_duplicate_history.py]: Bah. Sorry about that. Removed the typo. Thanks!

**Comment by rossmechanic** [comment]
looks like you need to autoformat this with `black`

**Comment by craigmaloney** [comment]
Actually, I ran it with black in #606. I can go back and run it again with this branch, but that branch includes this commit (that'll teach me to commit to master on my branch. :grin: )

**Comment by rossmechanic** [comment]
@craigmaloney that branch failed too though :( can you run it with this branch, we'll merge this, and then I'll look at the other? Thanks

**Comment by craigmaloney** [comment]
which part failed? There were two other files that I didn't commit because I hadn't changed them:

        modified:   simple_history/tests/tests/test_manager.py
        modified:   simple_history/tests/tests/test_models.py

If you want I can make a separate branch to commit those. I'm a little reluctant to merge them in with my work. :grin:

**Comment by rossmechanic** [comment]
If you click on the Travis CI build `continuous-integration/travis-ci/pr` details, you can see which build fails, and we can merge and changes with failed builds. Can you create a branch with the changes in this PR, run black on them, and then I'll merge them. Then we can look at the other PR

**Comment by rossmechanic** [comment]
Looks like issue on the other PR is also a black formatting issue

**Comment by rossmechanic** [comment]
Ah, I see. The issue you saw is probably because we don't have `black` pinned to a version. I'll fix that

**Pull request merged**